### ISSUE TITLE
Got loss nan at last iteration

### ISSUE THREAD

**Issue by shiva1393** [mravanelli/pytorch-kaldi#112]
ep=22 tr=['train_clean_100'] loss=0.671 err=0.214 valid=dev_clean loss=0.693 err=0.212 lr_architecture1=6.25e-06 lr_architecture2=6.25e-06 time(s)=17255
ep=23 tr=['train_clean_100'] loss=nan err=0.218 valid=dev_clean loss=nan err=0.911 lr_architecture1=3.125e-06 lr_architecture2=3.125e-06 time(s)=17416

Here iam using fbank librispeech data set.it trained well upto 22 epochs then i got nan values.before i used same network for fmllr and got results which described in paper.
I did any mistake?

**Comment by mravanelli** [comment]
Hi,
you can probably try to activate gradient clipping (by uncommenting lines
216-218 of core.py). Does it help?

Mirco
